File: src/main/java/pkg/exoad/swiggie/stx/SwWrapper.java
package pkg.exoad.swiggie.stx;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(fluent = true) @Getter @Setter public class SwWrapper<T>
{
	public static <T> SwWrapper<T> acquire(T value)
	{
		return new SwWrapper<>(value);
	}

	private T value;

	private SwWrapper(T value)
	{
		this.value = value;
	}
}

File: src/main/java/pkg/exoad/swiggie/stx/SwWrapper.kt
package pkg.exoad.swiggie.stx



class SwWrapper<T> private constructor(var value:T)
{
	fun value():T
	{
		return value
	}
	
	fun value(value:T)
	{
		this.value=value
	}
	companion object
	{
		@JvmStatic
		fun <T> acquire(value:T):SwWrapper<T>
		{
			return SwWrapper(value)
		}
	}
}

